Change advection-diffusion "Convection" keyword to "[solute-name] Convection"
in GUI with corresponding priority change in solver's .sif file parser.

Index: elmerfem/ElmerGUI/Application/edf-extra/advection-diffusion.xml
===================================================================
--- elmerfem.orig/ElmerGUI/Application/edf-extra/advection-diffusion.xml	2010-07-21 18:08:18.000000000 -0400
+++ elmerfem/ElmerGUI/Application/edf-extra/advection-diffusion.xml	2010-07-21 18:09:59.000000000 -0400
@@ -24,6 +24,7 @@
         <Parameter Widget="Combo" >
             <Type> String </Type>
             <Name> Convection </Name>
+            <SifName> Concentration Convection </Name>
             <Item Type="Active" > <Name> None </Name> </Item> <Item>
                <Name> Constant </Name>
                <Activate> /Advection Diffusion Equation/Equation/Convection Velocity 1 </Activate>
@@ -34,19 +35,19 @@
          </Parameter>
          <Parameter Widget="Edit" Enabled="False" Visible="False">
             <Name> Convection Velocity 1</Name>
-            <SifName> Convection velocity 1</SifName>
+            <SifName> Concentration Convection velocity 1</SifName>
             <Whatis> Convection velocity for 'Constant' convection model. Despite its association, may be space and time varying. </Whatis>
             <StatusTip> Convection velocity </StatusTip>
          </Parameter>
          <Parameter Widget="Edit"  Enabled="False" Visible="False">
             <Name> Convection Velocity 2</Name>
-            <SifName> Convection velocity 2</SifName>
+            <SifName> Concentration Convection velocity 2</SifName>
             <Whatis> Convection velocity for 'Constant' convection model. Despite its association, may be space and time varying. </Whatis>
             
          </Parameter>
          <Parameter Widget="Edit"  Enabled="False" Visible="False">
             <Name> Convection Velocity 3</Name>
-            <SifName> Convection velocity 3</SifName>
+            <SifName> Concentration Convection velocity 3</SifName>
             <Whatis> Convection velocity for 'Constant' convection model. Despite its association, may be space and time varying. </Whatis>
             <StatusTip> Convection velocity </StatusTip>
          </Parameter>
Index: elmerfem/fem/src/AdvectionDiffusion.src
===================================================================
--- elmerfem.orig/fem/src/AdvectionDiffusion.src	2010-07-21 18:08:18.000000000 -0400
+++ elmerfem/fem/src/AdvectionDiffusion.src	2010-07-21 18:12:42.000000000 -0400
@@ -341,11 +341,11 @@
            eq_id = ListGetInteger( Model % Bodies(body_id) % Values,'Equation', &
                       minv=1,maxv=Model % NumberOfEquations )
            ConvectionFlag = ListGetString( Model % Equations(eq_id) % Values, &
-                         'Convection', GotIt )
+                         TRIM(ComponentName(Solver % Variable)) // &
+                         ' Convection', GotIt )
            IF ( .NOT. GotIt ) &
              ConvectionFlag = ListGetString( Model % Equations(eq_id) % Values, &
-                           TRIM(ComponentName(Solver % Variable)) // &
-                           ' Convection', GotIt )
+                           'Convection', GotIt )
 
            ScaledToSolubility = .FALSE.
            ConcentrationUnits = ListGetString( Model % Equations(eq_id) % &
@@ -380,11 +380,11 @@
              END IF
            END IF
 
-           ConvectName = ListGetString( Material, &
-               'Convection Field Variable', GotIt )
+           ConvectName = ListGetString( Model % Equations(eq_id) % Values, &
+               TRIM(ComponentName(Solver % Variable)) // &
+               ' Convection Field Variable', GotIt )
            IF ( .NOT. GotIt ) &
-             ConvectName = ListGetString( Model % Equations(eq_id) % Values, &
-                 TRIM(ComponentName(Solver % Variable)) // &
+             ConvectName = ListGetString( Material, &
                  'Convection Field Variable', GotIt )
            IF ( GotIt ) THEN
              FlowSol => VariableGet( Solver % Mesh % Variables, &
@@ -534,21 +534,21 @@
 !------------------------------------------------------------------------------
 
          IF ( ConvectionFlag == 'constant' ) THEN
-           U = ListGetReal( Material,'Convection Velocity 1',n,NodeIndexes, GotIt)
+           U = ListGetReal( Model % Equations(eq_id) % Values, &
+                TRIM(ComponentName(Solver % Variable)) // &
+                ' Convection Velocity 1',n,NodeIndexes )
            IF ( .NOT. GotIt ) &
-             U = ListGetReal( Model % Equations(eq_id) % Values, &
-                  TRIM(ComponentName(Solver % Variable)) // &
-                  ' Convection Velocity 1',n,NodeIndexes )
-           V = ListGetReal( Material,'Convection Velocity 2',n,NodeIndexes, GotIt)
+             U = ListGetReal( Material,'Convection Velocity 1',n,NodeIndexes, GotIt)
+           V = ListGetReal( Model % Equations(eq_id) % Values, &
+                TRIM(ComponentName(Solver % Variable)) // &
+                ' Convection Velocity 2',n,NodeIndexes )
            IF ( .NOT. GotIt ) &
-             V = ListGetReal( Model % Equations(eq_id) % Values, &
-                  TRIM(ComponentName(Solver % Variable)) // &
-                  ' Convection Velocity 2',n,NodeIndexes )
-           W = ListGetReal( Material,'Convection Velocity 3',n,NodeIndexes, GotIt)
+             V = ListGetReal( Material,'Convection Velocity 2',n,NodeIndexes, GotIt)
+           W = ListGetReal( Model % Equations(eq_id) % Values, &
+                TRIM(ComponentName(Solver % Variable)) // &
+                ' Convection Velocity 3',n,NodeIndexes )
            IF ( .NOT. GotIt ) &
-             W = ListGetReal( Model % Equations(eq_id) % Values, &
-                  TRIM(ComponentName(Solver % Variable)) // &
-                  ' Convection Velocity 3',n,NodeIndexes )
+             W = ListGetReal( Material,'Convection Velocity 3',n,NodeIndexes, GotIt)
            IF (.NOT. AbsoluteMass) THEN
               C1 = Density
               DO i=1,3
